### PR TITLE
Implement editing of course instance plan grants

### PR DESCRIPTION
**Pull request by nwalters512**
This further preps us for the rollout of checked access to features by allowing site-wide admins to edit the plan grants associated with a course instance. We expect most plan grants to occur at the institution level, but course-instance-specific plan grants can be useful for one-offs (gift plans, or extended trials).

### CODE REVIEW

File: apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+onDocumentReady(() => {
+  document.querySelectorAll('.js-plan').forEach((plan) => {
+    const enabledCheckbox = plan.querySelector<HTMLInputElement>('.js-plan-enabled');
+    const enabledType = plan.querySelector<HTMLSelectElement>('.js-plan-type');
+    enabledCheckbox.addEventListener('change', () => {
+      enabledType.disabled = !enabledCheckbox.checked;
+    });
+  });
+});
```
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts:11]: This is an exact copy of `institutionAdminGeneralClient.ts`. We previously talked about bringing over `behaviors/enable-by-checkbox.ts` from PT; maybe it's time to do that now?

File: apps/prairielearn/src/ee/pages/institutionAdminCourseInstance/institutionAdminCourseInstance.ts [hunk 2]
```diff
@@ -40,15 +63,34 @@ router.get(
...
+      unsafe_course_instance_id: req.params.course_instance_id,
+    });
+
     if (req.body.__action === 'update_enrollment_limit') {
       await queryAsync(sql.update_enrollment_limit, {
         institution_id: req.params.institution_id,
-        course_instance_id: req.params.course_instance_id,
+        course_instance_id: course_instance.id,
         enrollment_limit: req.body.enrollment_limit || null,
       });
```
Comment (nwalters512) [inline apps/prairielearn/src/ee/pages/institutionAdminCourseInstance/institutionAdminCourseInstance.ts:79]: Now that we have `selectCourseInstanceAndCourseInstitution` above, we can probably omit the `institution_id` check here, as we've already verified that the course instance belongs to the institution.

File: apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts:11]: The same thing happened in PT where we started accumulating more and more of these until we added `enable-by-checkbox`. I don't have a firm opinion about when is the right time to bring it over.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/institutionAdminCourseInstanceClient.ts:11]: In the interest of shipping, I'll retain the duplication for now.

File: apps/prairielearn/src/ee/pages/institutionAdminCourseInstance/institutionAdminCourseInstance.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/ee/pages/institutionAdminCourseInstance/institutionAdminCourseInstance.ts]: I changed this in 63a4c48.

**Pull request merged**